From 0ee4a33d27ab261ae343338a6ca3f3c0d732e908 Mon Sep 17 00:00:00 2001
From: Bernhard R. Link <brlink@debian.org>
Date: Sat, 12 Mar 2011 16:22:41 +0100
Subject: more usual flags usage

Always give CFLAGS to the linker, always CPPFLAGS to the compiler.
---
 Makefile |   15 ++++++++++-----
 1 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 8620a22..89ebbe8 100644
--- a/Makefile
+++ b/Makefile
@@ -27,6 +27,8 @@ SHCFLAGS  = -fPIC -fvisibility=hidden
 #SHLDFLAGS = --shared -Wl,--soname=libcliquer.so.1 -Wl,--version-script=libcliquer.map
 #SHCFLAGS  = -fPIC
 
+CPPFLAGS =
+
 ## Program options:
 
 # Enable long options for cliquer (eg. "cliquer --help"), comment out to disable.
@@ -43,22 +45,25 @@ libcliquer_OBJECTS = cliquer.o graph.o reorder.o
 
 
 testcases: testcases.o cliquer.o graph.o reorder.o
-	$(CC) $(LDFLAGS) -o $@ testcases.o cliquer.o graph.o reorder.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ testcases.o cliquer.o graph.o reorder.o
 
 libcliquer.so: $(libcliquer_OBJECTS) libcliquer.map
-	$(CC) $(LDFLAGS) $(SHCFLAGS) $(SHLDFLAGS) -o $@ $(libcliquer_OBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SHCFLAGS) $(SHLDFLAGS) -o $@ $(libcliquer_OBJECTS)
 
 cliquer: cl.o libcliquer.so
-	$(CC) $(LDFLAGS) -o $@ cl.o -L. -lcliquer
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ cl.o -L. -lcliquer
 
 
 cl.o testcases.o cliquer.o graph.o reorder.o: cliquer.h set.h graph.h misc.h reorder.h Makefile cliquerconf.h
 
 cl.o: cl.c
-	$(CC) $(CFLAGS) $(LONGOPTS) -o $@ -c $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LONGOPTS) -o $@ -c $<
 
 $(libcliquer_OBJECTS): %.o: %.c
-	$(CC) $(CFLAGS) $(SHCFLAGS) $(LONGOPTS) -o $@ -c $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(SHCFLAGS) -o $@ -c $<
+
+testcases.o: testcases.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -o $@ -c $<
 
 clean:
 	rm -f *.o *~ cliquer testcases *.so*
-- 
1.5.6.5

